graphsdk/src/androidTest/java/com/microsoft/graph/core/BaseClientTests.java

```diff
@@ -0,0 +1,122 @@
+package com.microsoft.graph.core;
+
+import android.test.AndroidTestCase;
+
+import com.microsoft.graph.authentication.IAuthenticationProvider;
+import com.microsoft.graph.authentication.MockAuthenticationProvider;
+import com.microsoft.graph.concurrency.IExecutors;
+import com.microsoft.graph.concurrency.MockExecutors;
+import com.microsoft.graph.http.IHttpProvider;
+import com.microsoft.graph.http.MockHttpProvider;
+import com.microsoft.graph.logger.ILogger;
+import com.microsoft.graph.logger.MockLogger;
+import com.microsoft.graph.serializer.ISerializer;
+import com.microsoft.graph.serializer.MockSerializer;
+
+/**
+ * Test cases for {@see BaseClient}
+ */
+public class BaseClientTests extends AndroidTestCase {
+
+    public static final String DEFAULT_GRAPH_ENDPOINT = "https://graph.microsoft.com/v1.0";
+    private String mEndpoint;
+    private BaseClient baseClient;
+    private IAuthenticationProvider mAuthenticationProvider;
+    private IExecutors mExecutors;
+    private IHttpProvider mHttpProvider;
+    private ILogger mLogger;
+    private ISerializer mSerializer;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        baseClient = new BaseClient() {
+            @Override
+            public String getServiceRoot() {
+                if(mEndpoint == null) {
+                    mEndpoint = DEFAULT_GRAPH_ENDPOINT;
+                }
+                return mEndpoint;
+            }
+
+            @Override
+            public void setServiceRoot(String value) {
+                mEndpoint = value;
+            }
+        };
+        mAuthenticationProvider = new MockAuthenticationProvider();
+        mExecutors = new MockExecutors();
+        mHttpProvider = new MockHttpProvider();
+        mLogger = new MockLogger();
+        mSerializer = new MockSerializer(null, "");
+    }
+
+    public void testNotNull() {
+        assertNotNull(baseClient);
+        assertNotNull(mAuthenticationProvider);
+        assertNotNull(mExecutors);
+        assertNotNull(mHttpProvider);
+        assertNotNull(mLogger);
+        assertNotNull(mSerializer);
+    }
+
+    public void testValidate() {
```

Comment:
Break this into 2 tests, then just assert.fail() if the expected exception is not thrown